OpenKh.Command.IdxImg/EpicGamesAssets.cs

```diff
@@ -0,0 +1,173 @@
+                public bool DoNotExtractAgain { get; set; }
+
+                protected int OnExecute(CommandLineApplication app)
+                {
+                    var directory = Path.GetDirectoryName(InputHed);
+                    var filter = Path.GetFileName(InputHed);
+
+                    Directory
+                        .GetFiles(string.IsNullOrEmpty(directory) ? "." : directory, filter)
+                        .Where(x => x.EndsWith(".hed"))
+                        .AsParallel()
+                        .ForAll(inputHed => Extract(inputHed, OutputDir));
+
+                    return 0;
+                }
+
+                protected void Extract(string inputHed, string output)
+                {
+                    var outputDir = output ?? Path.GetFileNameWithoutExtension(inputHed);
+                    using var hedStream = File.OpenRead(inputHed);
+                    using var img = File.OpenRead(Path.ChangeExtension(inputHed, "pkg"));
+
+                    foreach (var entry in Hed.Read(hedStream))
+                    {
+                        var hash = EpicGamesAssets.ToString(entry.MD5);
+                        if (!Names.TryGetValue(hash, out var fileName))
+                            fileName = hash;
+
+                        var outputFileName = Path.Combine(outputDir, fileName);
+                        if (DoNotExtractAgain && File.Exists(outputFileName))
+                            continue;
+
+                        Console.WriteLine(fileName);
+                        CreateDirectoryForFile(outputFileName);
+
+                        File.Create(outputFileName).Using(stream => stream.Write(img.SetPosition(entry.Offset).ReadBytes(entry.DataLength)));
+
+                        var hdAsset = new EgsHdAsset(img.SetPosition(entry.Offset));
+                        File.Create(outputFileName).Using(stream => stream.Write(hdAsset.ReadData()));
+
+                        foreach (var asset in hdAsset.Assets)
+                        {
+                            var outputFileNameRemastered = outputFileName + asset;
+                            Console.WriteLine(outputFileNameRemastered);
+                            CreateDirectoryForFile(outputFileNameRemastered);
+
+                            var assetData = hdAsset.ReadAsset(asset);
+                            File.Create(outputFileNameRemastered).Using(stream => stream.Write(assetData));
+                        }
+                    }
+                }
+
+                private static void CreateDirectoryForFile(string fileName)
+                {
+                    var directoryName = Path.GetDirectoryName(fileName);
+                    if (!Directory.Exists(directoryName))
+                        Directory.CreateDirectory(directoryName);
+                }
+            }
+
+            //private class ListCommand
```

Comment:
Leftover copy from the KH2 command?

Reply:
Right! Lemme fix that once I am back home